cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -379,7 +385,11 @@ globalCommand commands = CommandUI {
 
       ,option [] ["store-dir", "storedir"]
          "The location of the build store"
-         globalStoreDir (\v flags -> flags { globalStoreDir = v })
+         globalStoreDir (\v flags ->
+            case sequence $ fmap readAbsoluteFilePath v of
+                Just flag -> flags { globalStoreDir = flag }
+                Nothing -> error "expected an absolute directory name for --store-dir"
```

Comment:
`error` isn't the way to report errors to the user.  `error` is only acceptable for aborting in impossible cases, like you know that a list is non-empty, so in case of empty list you will `error` out.

There should be a mechanism in place to properly report errors in the command line options